### PR TITLE
treehouses memory <total|used|free> <-g|-m> (fixes #446)

### LINKED ISSUE #446: change output format for treehouses memory
```
root@treehouses:~/cli/modules# treehouses memory
Your rpi has 1986024 bytes of total memory with 681720 bytes used and 1304052 bytes avalaible

```
May need the memory in a readable format.

Reference:
```
root@treehouses:~/cli/modules# free -h
              total        used        free      shared  buff/cache   available
Mem:          1.9Gi       210Mi       1.2Gi       8.0Mi       455Mi       1.6Gi
Swap:            0B          0B          0B
```

### PR DESCRIPTION
**Pull request by phamduchongan93**
This update will add `'-g'` and `'-m'` argument for `treehouses memory [total|used|free]`. It also fix #446.

### CODE REVIEW

**Comment by dogi** [comment]
```
root@treehouses:~/cli# ./cli.sh memory -g
Your rpi has 3999784 bytes of total memory with 849756 bytes used and 3075 bytes avalaible
```

**Comment by dogi** [comment]
`./cli.sh help memory`

**Comment by pattanawadee88** [comment]
```
root@pattanawadee88:~/cli# ./cli.sh memory free total
3265
root@pattanawadee88:~/cli# ./cli.sh memory free g
3266
root@pattanawadee88:~/cli# ./cli.sh memory free m
3266
root@pattanawadee88:~/cli# ./cli.sh memory used m
654908
root@pattanawadee88:~/cli# ./cli.sh memory help
Your rpi has 3999784 bytes of total memory with 644928 bytes used and 3275 bytes avalaible
```

File: modules/memory.sh [hunk 1]
```diff
@@ -1,39 +1,82 @@
...
-  
-  echo "Your rpi has $((t)) bytes of total memory with $((ubc)) bytes used and $((f)) bytes avalaible"
-}
-
-function memory_total {
-  t=$(free | grep -i Mem | awk '{printf $2}')
+  option=$1
+  case $option in
+    '-g')
+      echo "Your rpi has $t gigabytes of total memory with $ubc gigabytes used and $f gigabytes avalaible"
```
Comment (pattanawadee88) [inline modules/memory.sh]: misspelling "available"

File: modules/memory.sh [hunk 2]
```diff
@@ -1,39 +1,82 @@
...
+      ;;
+    '-m')
+      echo "Your rpi has $t megabytes of total memory with $ubc megabytes used and $f megabytes avalaible"
```
Comment (pattanawadee88) [inline modules/memory.sh]: misspelling "available"

File: modules/memory.sh [hunk 3]
```diff
@@ -1,39 +1,82 @@
...
+      ;;
+    *)
+      echo "Your rpi has $t megabytes of total memory with $ubc megabytes used and $f megabytes avalaible"
```
Comment (pattanawadee88) [inline modules/memory.sh]: misspelling "available"

**Comment by phamduchongan93** [comment]
Previously the memory script was tested and return the wrong result with option -g since `free -h` command returns the wrong output.
The work around is to calculate from megabyte to gigabyte.
```
t_M=$(free -m | grep -i Mem | awk '{printf $2}'
t=$(echo "scale=2;$t_M/1024" | bc)
```
t will return value with gigabyte. The **issue** right now is we doesn't have `bc` command installed on the treehouses image.

File: modules/memory.sh [hunk 4]
```diff
@@ -1,55 +1,127 @@
...
-  ubc=$((u+bc))
+  option=$1
+  case $option in 
+    '-g')
+      u_M=$(free -m | grep -i Mem | awk '{printf $3}')
+      u=$(echo "scale=2;$u_M/1024" | bc)
+      bc_M=$(free -m | grep -i Mem | awk '{printf $6}')
+      bc=$(echo "scale=2;$bc_M/1024" | bc)
+      ubc=u+bc|bc
+      #ubc=$( (u+bc) | bc )
```
Comment (CalebProvost) [inline modules/memory.sh]: -g failure:
```
$ ./cli.sh memory
Your rpi has 3999784 bytes of total memory with 887628 bytes used and 3111444 bytes avalaible
$./cli.sh memory -g
Your rpi has 3.81 gigabytes of total memory with  gigabytes used and 2.96 gigabytes avalaible
```
The results are missing for memory used in gigabyte argument.
Cause is `ubc=u+bc|bc` should be instead be `$(echo $u + $bc | bc)`

Comment (phamduchongan93) [reply modules/memory.sh]: I changed the code as you recommended. It should be working now.

**Comment by irisb1701** [comment]
```
Your rpi has 946664 bytes of total memory with 597304 bytes used and 348908 bytes available
root@treehouses:/home/pi/cli# ./cli.sh memory -g
Your rpi has .90 gigabytes of total memory with .56 gigabytes used and .33 gigabytes available
root@treehouses:/home/pi/cli# ./cli.sh memory -m
Your rpi has 924 megabytes of total memory with 583 megabytes used and 340 megabytes available
root@treehouses:/home/pi/cli# ./cli.sh memory free 
350924
root@treehouses:/home/pi/cli# ./cli.sh memory free -m
342
root@treehouses:/home/pi/cli# ./cli.sh memory free -g
.33
root@treehouses:/home/pi/cli# ./cli.sh memory used
595864
root@treehouses:/home/pi/cli# ./cli.sh memory used -m
580
root@treehouses:/home/pi/cli# ./cli.sh memory used -g
.55
root@treehouses:/home/pi/cli# 
```
Works perfect for me too, just needs bc. @dogi should we include bc in builder?

**Pull request merged**